fighthealthinsurance/ml/ml_models.py

```
@@ -1577,26 +1579,58 @@ async def __infer(
                 ) as response:
                     logger.debug(f"Got response {response}...")
+                    # Raise ClientResponseError for HTTP error status codes (4xx, 5xx)
+                    # This allows subclasses to catch and handle specific errors like 429
+                    try:
+                        response.raise_for_status()
+                    except aiohttp.ClientResponseError as e:
+                        # Log the response body for debugging 400 errors
+                        try:
+                            error_body = await response.text()
+                            logger.error(f"HTTP {e.status} error from {api_base} for model {model}: {error_body}")
+                        except:
```

Ruff (0.14.10)

Do not use bare `except`

(E722)